spark/src/test/scala/org/apache/comet/CometExpressionSuite.scala

```diff
@@ -617,6 +618,136 @@ class CometExpressionSuite extends CometTestBase with AdaptiveSparkPlanHelper {
     }
   }
 
+  test("rlike simple case") {
+    val table = "rlike_names"
+    val gen = new DataGenerator(new Random(42))
+    Seq(false, true).foreach { withDictionary =>
+      val data = Seq("James Smith", "Michael Rose", "Rames Rose", "Rames rose") ++
+        gen.generateStrings(100, "rames Rose", 12)
```

Comment:
I think we need enough number of repeated string in order to make a dictionary. Right now, it has a low chance due to the randomness...

Reply:
@kazuyukitanimura I updated the test and have now added dictionary support